tasks/task_manager.cc

```diff
@@ -180,17 +181,18 @@ void task_manager::task::start() {
     }
     _impl->_status.start_time = db_clock::now();
 
+    auto module = _impl->_module;
+    auto gate_holder = module->async_gate().hold();
+
     try {
         // Background fiber does not capture task ptr, so the task can be unregistered and destroyed independently in the foreground.
         // After the ttl expires, the task id will be used to unregister the task if that didn't happen in any other way.
-        (void)with_gate(_impl->_module->async_gate(), [f = done(), module = _impl->_module, id = id()] () mutable {
```

Comment:
I think that `gate_closed_exception` is never really passed to `finish_failed` (it would do be handled if so).
It rather creates a future that would run in the background here, finishes with the `start` method and then at some point reactor gets back to this and throws but nothing handles it.

So, I suggest moving 
```
auto gate_holder = module->async_gate().hold();
```
under `try` and allow the task to record that the gate was closed

Reply:
I tried it and it didn't work.
Note that with_gate throws gate_closed_exception is the gate is closed.

Reply:
The problem that appears when `hold()` is called outside of `try` is that a task will be created, registered but never finished and unregistered from task manager.

But this can happen on shutdown only anyway, so I guess it's ok

Reply:
moved gate holding to make_task time in v3